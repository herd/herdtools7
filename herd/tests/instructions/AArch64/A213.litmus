AArch64 A213
(* SUBS test, uses CSET *)

{
    0: X1 = 2;
    0: X2 = 2;
}

P0 ;
SUBS W0, W1, W2;
CSET W3, MI; (* Fetch N *)
CSET W4, EQ; (* Fetch Z *)
CSET W5, CS; (* Fetch C *)
CSET W6, VS; (* Fetch V *)

forall (
    0: X3 = 0 /\
    0: X4 = 1 /\
    0: X5 = 1 /\
    0: X6 = 0
)
