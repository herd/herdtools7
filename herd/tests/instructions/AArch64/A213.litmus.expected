Test A213 Required
States 1
0:X3=0; 0:X4=1; 0:X5=1; 0:X6=0;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (0:X3=0 /\ 0:X4=1 /\ 0:X5=1 /\ 0:X6=0)
Observation A213 Always 1 0
Hash=072968b91d2e260c66dc1061be0d0e8c

